pkg/aws/instance_info_test.go

```diff
@@ -337,6 +397,17 @@ func TestSyntheticInstanceInfo(t *testing.T) {
 				InstanceType: "<multiple>",
 				VCPU:         4,
 				Memory:       17179869184,
+				Architecture: "amd64",
+			},
+		},
+		{
+			name:          "one type with arm64",
+			instanceTypes: []string{"m6g.xlarge"},
+			expectedInstance: Instance{
+				InstanceType: "m6g.xlarge",
+				VCPU:         4,
+				Memory:       17179869184,
+				Architecture: "arm64",
```

Comment:
What about if we add multiple instances types of different arch?

Reply:
Please don't 😆

Reply:
I understand we can't/won't support that, but will it error out or will it just go to the first type found? I wonder how hard it would be to debug if done by accident basically.

Reply:
Ok, I guess that is good enough for now, the UX validation/improvements should be done anyway closer to the operators (us). I just wanted to make sure it doesn't fail only when AWS tries to create an instance from an ASG or so.

Reply:
It leaves the question what happens when the AMI and first instance type are fine but the second instance type in the pool (e.g. spot for us) doesn't work with the AMI. Will need to check if AWS already blocks this for us.

Reply:
So it works fine for on-demand pools. The pool will have the first instance type with the correct AMI.

For spot pools it's not perfect yet. It takes the first types AMI but still leaves the all the instance types in the configuration. I guess when the type comes up it won't work. However, this would also happen now if you specify a type that's not compatible.

I would go ahead with this change keeping this in mind for spot pools and improve the UX (e.g. warning or error, docs) in a follow up.

Reply:
I also added a test for that below.